### DISCUSSION TITLE
Digitally Signed Wix4 Bundle Installer does not Install the Binaries

### DISCUSSION THREAD

**Discussion by rajanbabu033** [wixtoolset/issues#7721] [Questions]
We have created wix Bundle Installer. This is digitally signed.

This internally has following packages,

1. Dot Net Installation package(Exe)

2. 3rd party OpenTapWix package(msi) -> This is digitally signed

3. Our Application Installation Package(msi) ->This is digitally  signed

When I try to double click the bundle installer and click Install button, it try to open the Open Dialog instead of installing

Attached the log file and screenshot. Kindly help here.

Note : If bundle Installer is not digitally signed, then Installation is success.

![WixBundleInstaller_Error](https://github.com/wixtoolset/issues/assets/29473199/aebece84-f164-4bbb-aafb-9ef83e4ba379)

[ITS_Operator_Application_3.7.0.789_20230911103327.log](https://github.com/wixtoolset/issues/files/12577894/ITS_Operator_Application_3.7.0.789_20230911103327.log)

**Comment by robmen** [comment]
Sounds like you did not sign the bundle correctly. Double-check the documentation and make sure you followed all of the steps.

**Reply by rajanbabu033** [reply]
Yes. Following steps helps.
1.Detach BundleEngine from Bundle
2.Signed BundleEngine
3.Attach BundleEngine with Bundle
4.Signed Bundle now
we need to sign bundle Engine and bundle separately which works fine. I did it manually and works fine. But do we have any clean document to do this in build environment. We can not use wix.exe in build environment as using Wix4.

**Reply by robmen** [reply]
> But do we have any clean document to do this in build environment

Yes. Where have you looked so far?

**Comment by rajanbabu033** [comment]
Any good document to properly sign the bundle installer in wix4 environment in the build pipeline?

**Reply by robmen** [reply]
Where have you looked so far?

**Reply by rajanbabu033** [reply]
I was looking into https://wixtoolset.org/docs/tools/signing/

here not sure how below code detaching and signing BundleEngine. Also not sure how it gets SignBundleEngine.FullPath variable.

<Target Name="SignBundleEngine">
  <Message Importance="high" Text="SignBundleEngine: @(SignBundleEngine)" />
  <Exec Command='signtool.exe sign /v /f $(TestCertificate) /p password **%(SignBundleEngine.FullPath)**' />
</Target>

![image](https://github.com/wixtoolset/issues/assets/29473199/aea203e0-7524-4db8-a422-263e96be0e83)

**Reply by BMurri** [reply]
There is a target that calls the `SignBundleEngine` target after calling another target to detach the bundle engine from the bundle. That target creates the `SignBundleEngine` item(s) for you.

A target that runs after `SignBundleEngine` reattaches the bundle engine to the bundle and sets the `SignBundle` item(s) for you, then calls the `SignBundle` target. All that can be found in the `wix.sdk` nuget package.

**Reply by rajanbabu033** [reply]
Thank you!
https://wixtoolset.org/docs/tools/signing/
This document helps. It would be helpful if the document is more detailed.
